lib/chef/provider/service/openbsd_rcctl.rb

```diff
@@ -0,0 +1,157 @@
+# Author:: Joe Miller (<joeym@joeym.net>)
+# Copyright:: Copyright (c) 2014 Scott Bonds
+# Copyright:: Copyright (c) 2015 Joe Miller
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/mixin/command'
+require 'chef/mixin/shell_out'
+require 'chef/provider/service/init'
+require 'chef/resource/service'
+
+class Chef
+  class Provider
+    class Service
+      class OpenbsdRcctl < Chef::Provider::Service::Init
+
+        provides :service, platform_family: 'openbsd', override: true do |node|
+          Chef::Platform::ServiceHelpers.service_resource_providers.include?(:rcctl)
+        end
+
+        include Chef::Mixin::ShellOut
+
+        def initialize(new_resource, run_context)
+          super
+          new_resource.supports[:status] = true
+          new_resource.status_command("rcctl check #{new_resource.service_name}")
+          new_resource.start_command("rcctl start #{new_resource.service_name}")
+          new_resource.stop_command("rcctl stop #{new_resource.service_name}")
+          new_resource.restart_command("rcctl restart #{new_resource.service_name}")
+          new_resource.reload_command("rcctl reload #{new_resource.service_name}")
+        end
+
+        def load_current_resource
+          @current_resource = Chef::Resource::Service.new(new_resource.name)
+          current_resource.service_name(new_resource.service_name)
+
+          determine_current_status!
+          determine_current_parameters!
+
+          current_resource.enabled is_enabled?
+        end
+
+        def define_resource_requirements
+          shared_resource_requirements
+
+          requirements.assert(:start, :enable, :reload, :restart, :status) do |a|
+            a.assertion { service_exists? }
+            a.failure_message Chef::Exceptions::Service, "#{new_resource}: rcctl does not recognize this service."
+          end
+        end
+
+        # Enable service and optionally set any additional variables such as command line flags
+        # that are stored in the `parameters` hash.
+        #
+        # As of OpenBSD 5.7 the following variables can be set (rcctl(8)):
+        #
+        #   flags, timeout, user.
+        #
+        # This code does not strictly enforce these variables since rcctl(8) will
+        # exit non-zero if you give it something it does not recognize.
+        #
+        def action_enable
+          if current_resource.enabled && !parameters_need_update?
```

Comment:
so if parameters_need_update is false then the current_resource is probably not 'enabled' -- at least not enabled correctly so we then need to fix it.

if you make that change to the semantics of current_resource.enabled, then you can probably stop overriding action_enable and just provide your enable_service sublcass method, which is the correct way to go rather than hacking up the superclass.